omod/src/main/java/org/openmrs/module/appointments/web/controller/AppointmentController.java

```diff
@@ -237,23 +259,42 @@ public AppointmentDefaultResponse getAppointmentByUuid(@RequestParam(value = "uu
     @ResponseBody
     public ResponseEntity<Object> editAppointment(@Valid @RequestBody AppointmentRequest appointmentRequest) {
         try {
-            boolean applyForAll = appointmentRequest.getApplyForAll() == null
-                    ? false : appointmentRequest.getApplyForAll();
+            final boolean isValidAppointmentRequest = appointmentRequestEditValidator.validate(appointmentRequest);
+            if (!isValidAppointmentRequest)
+                throw new APIException(appointmentRequestEditValidator.getError());
+            boolean applyForAll = appointmentRequest.requiresUpdateOfAllRecurringAppointments();
+            final String appointmentRequestUuid = appointmentRequest.getUuid();
             if (applyForAll) {
```

Comment:
we are not using clientTimeZone variable here. As part of the code refactoring i see we have already fetching  **appointmentRequest.getTimeZone();** in "AllAppointmentRecurringPatternMapper" class. I think we can handle this Timezone exception there itself.

Reply:
Moved this validation to "AllAppointmentRecurringPatternMapper"